client/app/models/contact.coffee

```diff
@@ -148,6 +148,26 @@ module.exports = class Contact extends Backbone.Model
         _.compact([pf, fn, mn, gn, sf]).join ' '
 
 
+    # We override toJSON to ensure the exported object is the a true object.
+    # Allow a options.toVCF boolean to sanitize the output before.
+    toJSON: ({toVCF} = {}) ->
+        data = super
+        data.datapoints = data.datapoints.toJSON()
+
+        return data unless toVCF
+
+        # Isolate in a try..catch block to go outside main thread and pop error
+        try
+            # We remove the BDay datapoint if it doesn't match ISO 8601
+            unless moment(data.bday, 'YYYY-MM-DD', true).isValid()
+                delete data['bday']
+                throw new Error "Malformed bday for #{data.id}"
```

Comment:
I agree, but we use in the _ModelView_ the [`saveas` method in a callback](https://github.com/cozy/cozy-contacts/blob/development/client/app/views/models/contact.coffee#L183-L185), and this break the console flow. To be sure the error is propagated, when need to do it in another thread in the JS VM, that's `try..catch`do.

Anyway, throwing an error, even if we catch it directly, is probably nicest than direclty call the `console.error`?